src/liboslexec/llvm_instance.cpp

```diff
@@ -320,9 +320,9 @@ BackendLLVM::llvm_type_groupdata()
         TypeDesc* types = &group().m_userdata_types[0];
         int* offsets    = &group().m_userdata_offsets[0];
         int sz          = (nuserdata + 3) & (~3);
-        fields.push_back(ll.type_array(ll.type_bool(), sz));
+        fields.push_back(ll.type_array(ll.type_int8(), sz));
```

Comment:
Can you talk to changing the type of the 
`bool GroupData::userdata_init_flags;`
to
`int8 GroupData::userdata_init_flags;`
?
Is it to ensure the size of the GroupData structure is consistent across platforms?

Reply:
`userdata_init_flags` is a variable used as a `char` in `osl_bind_interpolated_param` which can take 3 different values: not initialized (0), not found (1), and initialized (2).
Boolean values are treated as 2-state variables with llvm operators. Due to this limitation, it was impossible to implement the `osl_bind_interpolated_param` functionality in `build_interpolated_getter` with llvm operators. Cases 1 and 2 were treated the same way with the bool array.
To solve this problem, I changed the array type from bool to int8. This solution doesn't change the logic, since `userdata_init_flags` was already used as a `char` array.
That way I could implement `osl_bind_interpolated_param` with llvm operators for the `build_interpolated_getter` functionality.

Reply:
@Uedaki , excellent explanation.  Can you add a comment that it is representing 3 states and list them.
Follow on, should it just go to a vanilla int vs int8?  Do we gain much by using 1 byte vs 4 bytes, not sure if alignment rules would cause the following variable to start on 4 byte boundary anyway.